diagnostic_updater/include/diagnostic_updater/diagnostic_updater.hpp

```diff
@@ -370,11 +370,45 @@ class Updater : public DiagnosticTaskVector
    * \param h Node handle from which to get the diagnostic_period
    * parameter.
    */
-  explicit Updater(rclcpp::Node::SharedPtr node = rclcpp::Node::make_shared("diagnostics_updater"))
```

Comment:
I deliberately decided against that because I feel there is no clear line of recommendation on how to write these nodes. 
I personally also feel changes in `TransformListener` should be made to require an external node.

Reply:
Sounds OK, thanks for the explanation.